src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1269,6 +1184,34 @@ public SolutionState WithAnalyzerReferences(IReadOnlyList<AnalyzerReference> ana
         return Branch(analyzerReferences: analyzerReferences);
     }
 
+    public DocumentId? GetFirstRelatedDocumentId(DocumentId documentId)
+    {
+        var projectState = this.GetProjectState(documentId.ProjectId);
+        if (projectState is null)
+            return null;
+
+        var documentState = projectState.DocumentStates.GetState(documentId);
+        if (documentState is null)
+            return null;
+
+        var filePath = documentState.FilePath;
+        if (string.IsNullOrEmpty(filePath))
+            return null;
+
+        foreach (var (_, siblingProjectState) in this.ProjectStates)
```

Comment:
>foreach [](http://example.com/codeflow?start=8&length=7)

would it be worth checking in _lazyFilePathToRelatedDocumentIds before doing the loop?

Reply:
sure!